migration/identity-migration/src/test/java/io/camunda/migration/identity/TenantMappingRuleMigrationHandlerTest.java

```diff
@@ -0,0 +1,170 @@
+import io.camunda.migration.identity.dto.Tenant;
+import io.camunda.migration.identity.dto.TenantMappingRule;
+import io.camunda.migration.identity.midentity.ManagementIdentityClient;
+import io.camunda.migration.identity.midentity.ManagementIdentityTransformer;
+import io.camunda.migration.identity.service.MappingService;
+import io.camunda.migration.identity.service.TenantService;
+import io.camunda.search.entities.TenantEntity;
+import io.camunda.zeebe.client.api.command.ProblemException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class TenantMappingRuleMigrationHandlerTest {
+  final ArgumentCaptor<Collection<MigrationStatusUpdateRequest>> migrationStatusCaptor =
+      ArgumentCaptor.forClass(Collection.class);
+  @Mock private ManagementIdentityClient managementIdentityClient;
+  @Mock private TenantService tenantService;
+  @Mock private MappingService mappingService;
+
+  private final ManagementIdentityTransformer managementIdentityTransformer =
+      new ManagementIdentityTransformer();
+  private TenantMappingRuleMigrationHandler migrationHandler;
+
+  @BeforeEach
+  void setUp() throws Exception {
+    MockitoAnnotations.openMocks(this).close();
+    // when(tenantService.fetchOrCreateTenant(any(), any())).thenReturn(null);
+    migrationHandler =
+        new TenantMappingRuleMigrationHandler(
+            managementIdentityClient, managementIdentityTransformer, tenantService, mappingService);
+  }
+
+  @AfterEach
+  void tearDown() {}
+
+  @Test
+  void stopWhenNoMoreRecords() {
+    // given
+    givenMappingRules();
+    when(tenantService.fetch(any(), any()))
+        .thenReturn(new TenantEntity(1L, "", "", Collections.emptySet()));
+
+    // when
+    migrationHandler.migrate();
+
+    // then
+    verify(managementIdentityClient, times(2)).fetchTenantMappingRules(anyInt());
+    verify(tenantService, times(4)).fetch(any(), any());
+    verify(tenantService, times(4)).assignMappingToTenant(any(), any());
+    verify(mappingService, times(2)).findOrCreateMapping(any(), any(), any());
+  }
+
+  @Test
+  void setErrorWhenMappingRuleCreationFailed() {
+    // given
+    givenMappingRules();
+    when(mappingService.findOrCreateMapping(any(), any(), any())).thenThrow(new RuntimeException());
+
+    // when
+    migrationHandler.migrate();
+
+    // then
+    verify(managementIdentityClient, times(2)).fetchTenantMappingRules(anyInt());
+    verify(tenantService, never()).fetch(any(), any());
+    verify(mappingService, times(2)).findOrCreateMapping(any(), any(), any());
+    verify(managementIdentityClient, times(2))
+        .updateMigrationStatus(migrationStatusCaptor.capture());
```

Comment:
:wrench: No argument captor needed, you can just verify the arguments directly here